src/Management/src/Endpoint/Actuators/HttpExchanges/Diagnostics/DiagnosticObserver.cs

```diff
@@ -3,27 +3,21 @@
 // See the LICENSE file in the project root for more information.
 
 using System.Diagnostics;
-using System.Reflection;
 using Microsoft.Extensions.Logging;
 
-namespace Steeltoe.Management.Endpoint.Actuators.Metrics.Diagnostics;
+namespace Steeltoe.Management.Endpoint.Actuators.HttpExchanges.Diagnostics;
 
-public abstract class DiagnosticObserver : IDiagnosticObserver
+internal abstract class DiagnosticObserver : IObserver<KeyValuePair<string, object?>>, IDisposable
 {
+    private const string ListenerName = "Microsoft.AspNetCore";
+    private const string ObserverName = "HttpExchangesDiagnosticObserver";
```

Comment:
This makes the type unusable for anything _except_ HttpExchanges. It creates a dependency between base and derived that shouldn't exist.

Reply:
I'm confused, you asked for these to be private fields... it doesn't matter to me either way, seems like something easily sorted if we come up with a need for another DiagnosticObserver

Reply:
Private fields, not constants. Their values should still be set by defined types.

Reply:
The problem is that tight coupling is being introduced here, which is a code smell. Another smell is that base types should not have knowledge of derived types. Sure, we can change it all later, but that doesn't mean introducing questionable code today is justified. The base and derived types could be unified into a single type, but I don't think that's worth the effort at this time.